doc/CompilationPython.md

```diff
@@ -0,0 +1,189 @@
+ 
+ At the moment, pypointmatcher can only be compiled with **version 2.5.0** of pybind11. To install the right version, run the following commands :
+ 
+ ```bash
+ cd pybind11
+ git checkout v2.5.0
+ ```
+ 
+Once this is done, run the following commands :
+
+```bash
+mkdir build && cd build
+cmake ..
+make check -j 4
+```
+
+This will both compile and run the pybind11 tests. Next, you can install pybind11 by running this command :
+
+```bash
+sudo make install
+```
+
+Once this is done, return to libpointmatcher's `/build` directory.
+
+You're now ready to proceed to the [configuration step](#configuration).
+
+### Adding pybind11 as a `git` submodule <a name="adding-pybind11"></a>
+
+An alternative to installing pybind11 on your system is to add its repository as a git submodule and create a symbolic link into the python directory. To do this, you will first need to clone the repository as a git submodule by running the following commands in your terminal from the contrib directory.
+
+```bash
+cd contrib
+git submodule add https://github.com/pybind/pybind11.git
+```
+
+This will add pybind11 as a git submodule of libpointmatcher into the contrib directory. Then, still from the contrib directory, run this command to create a symbolic link to pybind11 in the python directory :
+
+```bash
+ln -sr pybind11 ../python/pybind11
+```
+
+At the moment, pypointmatcher can only be compiled with **version 2.5.0** of pybind11. To install the right version, run the following commands :
+
+```bash
+cd pybind11
+git checkout v2.5.0
+```
+
+Before going any further, the `CMakeLists.txt` file from the `/python` directory needs to be modified as follows :
+
+```cmake
+#find_package(pybind11 2.5.0 REQUIRED)
+add_subdirectory(pybind11) # add this line
+
+# Everything in between remains unchanged
+
+#if (pybind11_FOUND)
+#    message(STATUS "pybind11 v${pybind11_VERSION}")
+
+    pybind11_add_module(pypointmatcher ${PYBIND11_SOURCES})
+
+    target_link_libraries(pypointmatcher
+                          PRIVATE
+                          pointmatcher
+                          ${EXTERNAL_LIBS})
+
+    add_dependencies(pypointmatcher pointmatcher)
+
+    install(TARGETS pypointmatcher LIBRARY DESTINATION ${PYTHON_INSTALL_TARGET})
+#else ()
+#	message(FATAL_ERROR "pybind11 is required! Please follow the \"Compiling \
+#libpointmatcher's with python\" instructions from the official libpointmatcher's documentation.")
+#endif ()
+```
+
+> ***IMPORTANT :*** When this method is used, it is very important to checkout the version **2.5.0** of pybind11 or it will be impossible to generate the build files.
+
+Once this is done, return to libpointmatcher's `/build` directory.
+
+You're now ready to proceed to the [configuration step](#configuration).
+
+## Configuring the variables <a name="configuration"></a>
+
+> ***Note :*** *It is recommended to create a virtual python environment before proceeding with the next steps.*
```

Comment:
Can you explain why is it recommended to use a virtual env?

Reply:
Yes, I can or I thought that I could put a link to a tutorial that would explain how and why creating a virtual environment. I think that [The Coding Train](https://youtu.be/nnhjvHYRsmM) tutorial (youtube) and [Real Python](https://realpython.com/python-virtual-environments-a-primer) tutorial are good ones. What do you think?